src/main.ts

```diff
@@ -67,6 +69,11 @@ export const markdocPreprocess = (options: Options = {}): PreprocessorGroup => {
   const componentsPath = options.components || "$lib/components";
   const layoutPath = options.layout;
   const allowComments = options.comments ?? true;
+  const processHeadings = options.headingIds ?? false;
+  // If passed `true`, use the default for slugging headings
+  // Otherwise, use the passed function
+  const headingSlugger =
+    typeof processHeadings === "boolean" ? slug : processHeadings;
```

Comment:
```suggestion
    // Use custom slugger if provided, or fallback to default
  const headingSlugger =
    typeof processHeadings === "function" ? processHeadings : slug;
```

Maybe this is a clearer way to express the intention of the logic?